### ISSUE TITLE
Minor: Funny week highlighting in mini calendar when using hidden_days 

### ISSUE THREAD

**Issue by embeh** [meeting-room-booking-system/mrbs-code#3788]
**Describe the bug**
Normally, the minicalendar shows the current week with a continuous visual indicator, using a left half-circle for the start of the week, a grey rectangle for the middle of the week, and a right half-circle for the end of the week.

This seems to be broken when using the `hidden_days` configuration. 
Then the minicalendar displays each day of the week separately with a full blue circle.

**To Reproduce**
1. Configure `$hidden_days = array(0,6);`
2. Open Week View
3. The mini calendar now looks like this:
![image](https://github.com/user-attachments/assets/e8760c01-baf8-46ff-995f-2bf08666c235)

**Expected behavior**
The display should be the same as for a full week, only with the left/right half-circle starting on the appropriate days.

**Screenshots**
![image](https://github.com/user-attachments/assets/cb217281-1dd5-4235-baf1-82a97e859a02)

**Server details (please complete the following information):**
 - MRBS version 1.11.6
 - Web server: Apache 2.4.62
 - OS: Ubuntu 20.04.06
 - PHP version: 8.3.14

**Comment by campbell-m** [comment]
I'll need to think about this.  It was done this way deliberately because when there are hidden days they are disabled for non-admins and it was just easier to highlight the allowed days.  But maybe something better could be done.

Were you expecting in your example above the range to start on the 9th and finish on the 13th?  If so, what would you expect to see if, for example, the hidden days were `[3, 5]`, splitting the week into three chunks?   I think if the range were to displayed then it would have to be the full 7 days, with some kind of appropriate styling for the hidden days.

**Comment by embeh** [comment]
> Were you expecting in your example above the range to start on the 9th and finish on the 13th? 

Yes, exactly, with a continuous highlight (no gaps) for 10, 11, and 12.

> If so, what would you expect to see if, for example, the hidden days were `[3, 5]`, splitting the week into three chunks? I think if the range were to be displayed then it would have to be the full 7 days, with some kind of appropriate styling for the hidden days.

Some thoughts on this:
- Hidden days being 3,5 seems to be a rare edge case, no? Whereas what I am showing in the screenshot is removal of the weekend, which might be much more common.
- For me, it would also be completely fine to only use a continuous range if the days are continuous, too, and revert to the "circle only" style otherwise.
- More generally speaking, my intuition would be: a.) Any two or more consecutive days would start with a left half-circle and end with a right half-circle, and any days "in between" would have a square background. b.) Only a single days would show a full circle
- If you were to always use a rectangular background instead of a circle all this discussion would go away automatically...
- Finally: This is only about eye candy, so not very high priority. It just looks odd on the mini calendar. So no real problem if you have better things to do.

**Comment by campbell-m** [comment]
> Hidden days being 3,5 seems to be a rare edge case, no?

Not as rare as you might think.  I know it's used by a number of sports clubs who only have use of their facilities on a few days a wek, eg Tuesdays and Thursdays.

Anyway, I agree it could look better but is not high priority so I'll think about it slowly in the background.
